PageObjects/homePageFns.js

```diff
@@ -23,4 +28,58 @@ export default class HomePage {
     async navigateToHomePage(url) {
         await this.page.goto(url)
     }
+
+    /**
+     * Select the respective item from search result cards
+     * @param {string} item -The item that has to be selected
+     */
+    async selectItemFromSearchResult(item) {
+        const totalSearchResult = await this.page.locator(searchResultTitle).textContent()
+
+        //if only 1 search result found directly clicking on add to cart btn
+        if (totalSearchResult === '1 Results') {
+            await this.page.getByText(addToCartText, { exact: true }).click()
+        }
+        //else checking for item name and then clicking
+        else {
+            const totalItemCount = await this.page.locator(itemCard).count()
+
+            for (let i = 0; i < totalItemCount; i++) {
+                if (await this.page.locator(itemCard).nth(i).locator(itemNameInCard).textContent() === item) {
+                    //if found click on add to cart
+                    await this.page.locator(itemCard).nth(i).getByText(addToCartText, { exact: true }).click()
+                }
+            }
```

Comment:
Refactor this code and make it more readable, divide the logic if required use local variable type to reuse repeated code

Reply:
Thanks for the feedback, I will refactor this.